README.md

```diff
@@ -1 +1,149 @@
-# ReactRailBackEnd
\ No newline at end of file
+<a name="readme-top"></a>
+
+# Hello Rails Back End
+
+# 📗 Table of Contents
+
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+  - [Key Features](#key-features)
+  - [🚀 Live Demo](#live-demo)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+- [Usage](#usage)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+# 📚 Hello Rails <a name="about-project"></a>
+
+**Hello-Rails-Back-End** in this repo, i am building the rails api.
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+<details>
+<summary>Backend</summary>
+  <ul>
+    <li><a href="https://www.postgresql.org/">PostgreSQL</a></li>
+  </ul>
+  <ul>
+    <li><a href="https://www.ruby.org/">Ruby</a></li>
+  </ul>
+  <ul>
+    <li><a href="https://www.rubyonrails.org/">Ruby on Rails</a></li>
+  </ul>
+</details>
+
+### Link to React Front End
+You can find the React app repository [here](https://github.com/ibugithub/ReactRailFrontEnd/)
+
+### Key Features <a name="key-features"></a>
+
+- **Initialize javascript webpack**
+- **Installed node dependencies**
+- **Create a table with different greetings**
+- **Create an api from rails**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project you need:
+
+[Install Ruby](https://www.ruby-lang.org/en/documentation/installation/)
+
+### Setup
+
+Clone or download the repository to your local machine.
+
+```sh
+  cd my-folder
+  git clone https://github.com/ibugithub/ReactRailBackEnd
+```
+
+Open your terminal and navigate to the cloned project directory.
+
+### Install
+
+This project does not require additional dependencies, just Ruby interpreter. However, it uses RuboCop as a linter and you should set it up as well.
+
+```sh
+bundle install
+npm install
+```
+
+### Usage
+
+To run the project, navigate to the project directory and execute the following command:
+
+```sh
+  bundle install
+  bin/rails db:create db:migrate rails server
+
+  open http://localhost:3000/api/random_gretting
+```
+
+### Run tests
+
+To run tests, run the following command:
+
+Run the following command
+
+```sh
+rspec spec/
```

Comment:
- [ ] Kindly add the missing links to the subsections in the table of contents for a more descriptive readme file
![image](https://github.com/ibugithub/ReactRailBackEnd/assets/81932478/0da5c1c0-1e84-44b0-bb1c-0ad90043ce73)
 These are the `Install` and `Run tests` links

Reply:
@Munsa1 Hey man thanks so much .